### PR TITLE
[Site] Setup contact form

### LINKED ISSUE #31: [Site] Setup contact form
See https://www.eder-gmbh.de/kontaktformular/

- [ ] setup Franklin sheet for contact form
- [ ] style contact form

See https://www.hlx.live/developer/forms

### PR DESCRIPTION
**Pull request by franzhcs**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #31 

Known bugs / limitations:

- [x] The "Anrede" fields are not exactly how Eder displays them. I tried to finely tune them, but they still look different
- [x] The "Datenschutzerklärung" has two *. The original page has the mandatory asterisk on the label, not on the field. But if I remove it from the field, then it breaks the validation
- [ ] There's a layout bug that if you only put a value for "nachname" and press submit, the columns layout break because of the spacing introduced by the error message
- [x] The "* erforderliche Angaben" is not aligned to the right, despite being aligned correctly in the Onedrive Word
- [x] The "Datenschutzerklärung" link in the checkbox appears as it isn't bold - the styling is correct but the Roboto font is not currently used in the website, and thus the rendering tricks you
- [x] The "Wohnort" field in the original website has a "up/down" button, which I find quite weird so I didn't port it in my version
- [ ] There's a warning that "An invalid form control with name='' is not focusable" in the console. I am not sure why this is happening, but it doesn't look like anything serious.
- [ ] The shape of the checkbox is not the same from Eder - they use a different one with a different box, which I couldn't find
- [x] The textarea border is lightblue when selected, oddly enough is the only input with that behaviour
- [x] When hovered, the Anrede labels don't become red and buttons don't turn dark grey
- [x] When opened from mobile, the fields do not have width: 100% and thus don't appear stacked
- [x] Sidebar gets introduced at 900px in current Franklin site implementation, in original site it starts appearing at 768px.
- [x] Top headline for form needs to be in proper case rather than upper case with reduced font size.

Test URLs:
- Reference page: https://www.eder-gmbh.de/kontaktformular/
- Before: https://main--eder-group--hlxsites.hlx.page/kontaktformular
- After: https://issue-31--eder-group--hlxsites.hlx.page/kontaktformular

### CODE REVIEW

**Comment by mhaack** [comment]
Can you merge main branch into please.

File: scripts/scripts.js [hunk 1]
```diff
@@ -167,6 +167,34 @@ export function addFavIcon(href) {
   }
 }
 
+/**
+ * Returns true of the input string is a valid email address.
+ * @param {string} email An email address
+ */
+export function validateEmail(email) {
```
Comment (mhaack) [inline scripts/scripts.js:174]: The will only be needed by the form, lets move them there. No need to load the code all the time.

File: scripts/scripts.js [hunk 2]
```diff
@@ -167,6 +167,34 @@ export function addFavIcon(href) {
...
+  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(email);
+}
+
+/**
+ * Returns true of the input string is a valid phone number.
+ * @param {string} phone A phone number
+ */
+export function validatePhone(phone) {
```
Comment (mhaack) [inline scripts/scripts.js:183]: The will only be needed by the form, lets move them there. No need to load the code all the time.

File: scripts/scripts.js [hunk 3]
```diff
@@ -167,6 +167,34 @@ export function addFavIcon(href) {
...
+  const re = /^(?:\+|\d)[\d- ]{8,}$/;
+  return re.test(phone);
+}
+
+/**
+ * Returns true of the input string is a valid value for PLZ (5-digit number).
+ * @param {string} number A PLZ
+ */
+export function validatePLZ(number) {
```
Comment (mhaack) [inline scripts/scripts.js:192]: The will only be needed by the form, lets move them there. No need to load the code all the time.

File: blocks/form/form.css [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+  font-size: var(--type-heading-m-size);
+  font-weight: bold;
+  text-transform: uppercase;
+}
+
+main .form input,
+main .form textarea,
+main .form select {
+  font-size: 16px;
+  border: 1px solid #333;
```
Comment (mhaack) [inline blocks/form/form.css:27]: Use color variables.

File: blocks/form/form.css [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+  padding: 0 15px;
+  border-radius: 0;
+  height: 45px;
+  width: 100%;
+  box-sizing: border-box;
+  margin-top: 4px;
+  line-height: 1.4286;
+  color: #555;
```
Comment (mhaack) [inline blocks/form/form.css:35]: Use color variables.

File: blocks/form/form.css [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+  background-color: #fff;
```
Comment (mhaack) [inline blocks/form/form.css:36]: Use color variables.

File: blocks/form/form.css [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+  background: url('./checkbox-checked.svg');
+  background-size: contain;
+  content: ' ';
+}
+
+main .form .button {
+  cursor: pointer;
+  border-radius: 4px;
+  font-size: 18px;
+  color: #fff;
```
Comment (mhaack) [inline blocks/form/form.css:133]: Use color variables.

File: blocks/form/form.css [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+  background-size: contain;
+}
+
+main .form .form-checkbox-wrapper input[type='checkbox']:checked +label::before {
+  background: url('./checkbox-checked.svg');
+  background-size: contain;
+  content: ' ';
+}
+
+main .form .button {
```
Comment (mhaack) [inline blocks/form/form.css:129]: Is the form button that different from the default button? Only add differences here.

File: blocks/form/form.css [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+}
+
+main .form .radio-inner {
+  display: inline-block;
+  margin: 5px 10px;
+  padding: 0;
+}
+
+main .form .radio-inner label {
+  border: 1px solid #333;
```
Comment (mhaack) [inline blocks/form/form.css:167]: Use color variables.

File: blocks/form/form.css [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+  cursor: pointer;
+}
+
+main .form .field-wrapper.form-bold label {
+  font-weight: bold;
+}
+
+main .form .radio-inner input:checked~label {
+  color: #fff;
+  background-color: #de010a;
```
Comment (mhaack) [inline blocks/form/form.css:181]: Use color variables.

File: blocks/form/form.css [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+  border-color: #de010a;
```
Comment (mhaack) [inline blocks/form/form.css:182]: Use color variables.

File: blocks/form/form.css [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+main .form::after {
+  content: "";
+  display: table;
+  clear: both;
+}
+
+main .form .error-message {
+  list-style: none;
+  padding: 0;
+  color: #de010a;
```
Comment (mhaack) [inline blocks/form/form.css:194]: Use color variables.

File: blocks/form/form.css [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+  font-weight: 500;
+  top: -5px;
+  position: relative;
+  margin-bottom: 0;
+}
+
+::placeholder {
+  color: #999;
```
Comment (mhaack) [inline blocks/form/form.css:202]: Use color variables.

File: blocks/form/form.css (hunk 8 above)
Comment (franzhcs) [reply blocks/form/form.css:124]: Adjusted, kept only differences.

**Comment by franzhcs** [comment]
@mhaack  I performed the required changes.

**Comment by mhaack** [comment]
LGTM. I suggest to move the .svg files to the /images folder where all the oder images / SVG are.

**Comment by mhaack** [comment]
On the open items you mention in the PR, I think the following should be address. The others I would ignore.

- There's a layout bug that if you only put a value for "nachname" and press submit, the columns layout break because of the spacing introduced by the error message
This one I would try to fix.

- The "* erforderliche Angaben" is not aligned to the right, despite being aligned correctly in the Onedrive Word
Needs to be fixed in CSS, word alignments are ignored

- The "Datenschutzerklärung" link in the checkbox appears as it isn't bold - the styling is correct but the Roboto font is not currently used in the website, and thus the rendering tricks you
Double check Roboto font now is there

- The textarea border is lightblue when selected, oddly enough is the only input with that behaviour
This can be fixed in CSS with `:focus`

**Comment by asthabh23** [comment]
@franzhcs Few layout changes required:
- In mobile view (rather till 768px) the text boxes need to be 100% width:
Current:
Screenshot 2023-02-03 at 1 58 01 PM
Expected:
Screenshot 2023-02-03 at 1 58 27 PM

- On hovering the buttons, the button outline and text should turn red, can be handled with `:hover` in CSS.
Current:
Screenshot 2023-02-03 at 2 00 09 PM
Expected:
Screenshot 2023-02-03 at 2 01 23 PM

- Button on hover turns red to black with white text inside.  Can be fixed with `:hover` in CSS.
Current:
Screenshot 2023-02-03 at 2 01 58 PM
Expected:
Screenshot 2023-02-03 at 2 03 32 PM

- Sidebar gets introduced at 900px in current Franklin site implementation, in original site it starts appearing at 768px. 

- Top headline for form needs to be in proper case rather than upper case with reduced font size.
Current:
Screenshot 2023-02-03 at 2 08 27 PM
Expected:
Screenshot 2023-02-03 at 2 08 57 PM

**Comment by steinkopffp** [comment]
The Form Spacing regards label and the input fields are a little off same go's for the selection. Since we need in another case I think we can merge the current state

**Review by steinkopffp — APPROVED** [review]

**Pull request merged**